client/consensus/grandpa/aux_schema.go

```diff
@@ -179,15 +179,39 @@ func UpdateAuthoritySet[H comparable, N constraints.Unsigned](store AuxStore, se
 // We always keep around the justification for the best finalized block and overwrite it
 // as we finalize new blocks, this makes sure that we don't store useless justifications
 // but can always prove finality of the latest block.
-func UpdateBestJustification() {
-	// TODO impl when we have justification logic
-	panic("impl")
+func UpdateBestJustification[H HashI, N constraints.Unsigned, S comparable, ID constraints.Ordered, Header HeaderI[H, N]](store AuxStore, justification Justification[H, N, S, ID, Header]) error {
+	encodedJustificaiton, err := scale.Marshal(justification)
+	if err != nil {
+		return err
+	}
+
+	insert := []KeyValue{
+		{BEST_JUSTIFICATION, encodedJustificaiton},
```

Comment:
For consistency with our code, I would suggest writing as camel case instead. So `bestJustification` for private or `BestJustification` if it needs to be public to other packages